src/main/java/graphql/annotations/searchAlgorithms/SearchAlgorithm.java

```diff
@@ -0,0 +1,21 @@
+/**
+ * Copyright 2016 Yurii Rashkovskii
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ */
+package graphql.annotations.searchAlgorithms;
+
+import java.lang.reflect.Member;
+
+public interface SearchAlgorithm {
+    boolean search(Member member) throws CannotCastMemberException;
```

Comment:
it returns true if it exists?
maybe it would be right to change it to "isExists" or "isFound" or to keep it as search, but return the found object instead and null if not exists.
Because the name of the method does not imply it's functionallity